marklogic-data-hub-central/ui/src/components/queries/saving/save-query-modal/save-query-modal.tsx

```diff
@@ -0,0 +1,154 @@
+                setAllSearchFacets(greyedOptions.selectedFacets);
+                facets = {...facets,...greyedOptions.selectedFacets};
+                clearAllGreyFacets();
+                props.toggleApplyClicked(true);
+                props.toggleApply(false);
+                break;
+            case 2:
+                break;
+            case 3:
+                clearAllGreyFacets();
+                props.toggleApplyClicked(true);
+                props.toggleApply(false);
+        }
+
+        if(queryName.length > 0 ){
+            props.saveNewQuery(queryName, queryDescription, facets);
+        } else {
+            isQueryEmpty('error')
+        }
+    }
+
+    const handleChange = (event) => {
+        if (event.target.id === 'save-query-name') {
+           setQueryName(event.target.value);
+        }
+        if (event.target.id === 'save-query-description') {
+           setQueryDescription(event.target.value);
+        }
+    }
+
+    const unAppliedFacets = (e) => {
+        setRadioOptionClicked(e.target.value)
+    }
+
+
+    return (
+        <Modal
+            visible={true}
+            title={'Save Query'}
+            closable={true}
+            onCancel={() => onCancel()}
+            onOk={() => onOk()}
+            maskClosable={true}
+            footer={null}
+            data-testid='save-query-modal'
+        >
+            <Form
+                name="basic"
+                {...layout}
+            >
+                <Form.Item
+                    colon={false}
+                    label={<span className={styles.text}>
+                           Name:&nbsp;<span className={styles.asterisk}>*</span>&nbsp;
+                        </span>}
+                    labelAlign="left"
+                    validateStatus={queryEmpty}
+                    help={ queryEmpty === 'error' ? 'Query name is required' : ''}
+                    data-testid='save-query-name'
+                >
+                    <Input
+                        id="save-query-name"
+                        value={queryName}
+                        placeholder={'Enter query name'}
+                        onChange={handleChange}
+                    />
+                </Form.Item>
+                <Form.Item
+                    colon={false}
+                    label='Description:'
+                    labelAlign="left"
+                    data-testid='save-query-description'
+                >
+                    <Input
+                        id="save-query-description"
+                        value={queryDescription}
+                        onChange={handleChange}
+                        placeholder={'Enter query description'}
+                    />
+                </Form.Item>
+                {props.greyFacets.length > 0  && <Form.Item
+                    colon={false}
+                    label='Unapplied Facets:'
+                    labelAlign="left"
+                    data-testid='unapplied-radio-options'
```

Comment:
Same as above.

Reply:
@bsrikan removed all the data-testid references